artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/QueueControlImpl.java

```diff
@@ -1770,8 +1770,26 @@ public String listConsumersAsJSON() throws Exception {
 
             if (consumer instanceof ServerConsumer) {
                ServerConsumer serverConsumer = (ServerConsumer) consumer;
-
-               JsonObjectBuilder obj = JsonLoader.createObjectBuilder().add("consumerID", serverConsumer.getID()).add("connectionID", serverConsumer.getConnectionID().toString()).add("sessionID", serverConsumer.getSessionID()).add("browseOnly", serverConsumer.isBrowseOnly()).add("creationTime", serverConsumer.getCreationTime());
+               List<MessageReference> deliveringMessages = consumer.getDeliveringMessages();
+               long deliveringMessageSize = 0;
+               for (int i = 0; i < deliveringMessages.size(); i++) {
+                  MessageReference messageReference =  deliveringMessages.get(i);
+                  deliveringMessageSize += messageReference.getMessage().getEncodeSize();
```

Comment:
If we want this for metrics would be safer to keep this count as a metric on the consumer adding the size and removing it from the metric as messages are added or ack'd this would also save on the above mentioned npe / concurrent issue